### PR TITLE
add new param current_status to /api/order cancel

### PR DESCRIPTION
**Pull request by jerryfletcher21**
## What does this PR do?
This PR introduces a new optional parameter `current_status` that can be used when cancelling an order. The server checks if `current_status` is the correct status of the order and returns an error without cancelling the order if it is not.
This way a client can be sure to cancel an order just if the status is the one it specified.
Fixes backed and tests of #1311.

## Checklist before merging
- [X] Install [pre-commit](https://pre-commit.com/#installation) and initialize it: `pip install pre-commit`, then `pre-commit install`. Pre-commit installs git hooks that automatically check the codebase. If pre-commit fails when you commit your changes, please fix the problems it points out.

### CODE REVIEW

File: robosats/settings.py [hunk 1]
```diff
@@ -146,6 +146,9 @@
         }
     },
     "REDOC_DIST": "SIDECAR",
+    "ENUM_NAME_OVERRIDES": {
```
Comment (Reckless-Satoshi) [inline robosats/settings.py:149]: Aha! TIL :smile:

**Review by Reckless-Satoshi — COMMENTED** [review]
This PR is solid and complete. We will be merging #1326 instead given that it implements the frontend fixes and has integration tests as well. But a few things from this PR can be implemented there for the best of both worlds :smile: 

Makes sense to split the reward as discussed on #1326. Please post an invoice for 40K Sats (+ week expiration if possible) :rocket: 

Closing.

File: robosats/settings.py (hunk 1 above)
Comment (jerryfletcher21) [reply robosats/settings.py:149]: I also learned it today, well yesterday, while searching how to remove that warning ahahaha

**Comment by jerryfletcher21** [comment]
> This PR is solid and complete. We will be merging #1326 instead given that it implements the frontend fixes and has integration tests as well. But a few things from this PR can be implemented there for the best of both worlds 😄
> 
> Makes sense to split the reward as discussed on #1326. Please post an invoice for 40K Sats (+ week expiration if possible) 🚀
> 
> Closing.

Sure, perfectly fine with me, I left some comments in #1326

**Comment by jerryfletcher21** [comment]
@KoalaSat 

`lnbc400u1p58pu54sp5e7wgxuq60t0lxx9ps6dk87s67h4720tadnth92az6cek46vhalfspp5w2nk2lcwgajs6vq9699lg3gggd9n9w8rp3c3ace97ldlp527ghfsdqqxq9yn4qqcqzysrzjqdllz2m2fe9ukju5fdkjptcgehlkrv6xrswl7rgq4zrfws2d3yduwr8j0yqpv2sqqqqqqqlgqqqqp9sq2q9qxpqysgqmcngp2z0cl7x9rv0rmvxfxv4pu42v8hcy46rhfx9em0gr9tat779g34l5gwk7whfyw0dpl8asypc782c7pa8zkxte9pkxst6fwpm22qqzjqkhy`

**Comment by KoalaSat** [comment]
d4735a610cbd96616b0dde3196a9939bdfc18a182c26e0de9aaa60711702cc0a

**Pull request closed without merging**